frontend/components/hourInput.tsx

```diff
@@ -1,6 +1,62 @@
+import { ChangeEvent, FocusEvent, useState } from "react";
+
+export interface IWorkDuration {
+  hours: number;
+  minutes: number;
+}
+
+export const getWorkDuration = (s: string): IWorkDuration => {
+  let parts = s.split(":");
+  if (parts.length === 1) {
+    return {
+      hours: parseInt(parts[0]),
+      minutes: 0,
+    };
+  } else if (parts.length > 1) {
+    return {
+      hours: parseInt(parts[0]),
+      minutes: parseInt(parts[1]),
+    };
+  }
+
+  return {
+    hours: parseInt(s),
+    minutes: 0,
+  };
+};
+
+function pad(d) {
```

Comment:
```suggestion
function pad(d: number) {
```

I think we should add the type here, otherwise `d` would be an `any`. And nobody likes `any` 😄.